### ISSUE TITLE
[Rust] Implement new Concept Exercise: methods

### ISSUE THREAD

**Issue by efx** [exercism/rust#1066]
This issue describes how to implement the `methods` concept exercise for the Rust track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how to use methods in Rust. 

## Learning objectives

- know what the `impl` keyword means 
- know how to define methods on structs
- know how to read properties on an instance of a struct using `&self`
- know how to mutate properties on an instance of a struct using `&mut self`

## Out of scope

- generic methods
- destructors

## Concepts

- methods

## Prerequisites

- basics 
- structs-use 
- explicit-return

### Hints

- https://doc.rust-lang.org/stable/rust-by-example/fn/methods.html
- https://doc.rust-lang.org/book/ch05-03-method-syntax.html#defining-methods

### After

- https://doc.rust-lang.org/stable/rust-by-example/fn/methods.html
- https://doc.rust-lang.org/book/ch05-03-method-syntax.html#defining-methods

## Representer

No changes required

## Analyzer

No changes required

## Implementing

See our [implementation guide](https://github.com/exercism/v3/blob/master/languages/rust/howto/implementing-a-concept-exercise.md#how-to-implement-a-rust-concept-exercise) for guidance.
 You may want to make use of `boil_exercise.sh` in the `concept` directory.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by haraldreingruber** [comment]
Could it be that the [health-statistics](https://github.com/exercism/rust/blob/main/exercises/concept/health-statistics/src/lib.rs) exercise is already covering this concept?

**Comment by nahuakang** [comment]
I think @haraldreingruber's suggestion is valid since `health-statistics` seems like a perfect concept exercise for `methods`. My question is if we should write up the methods concept here?

@coriolinus I'll have to read up on how to create a concept directory and link an exercise to it but happy to claim this issue if no one has done/is doing work on it yet.

**Comment by coriolinus** [comment]
I wish that the structs concept required students to write at least one new struct on their own, and the methods concept required students to write at least one new method on their own. However, we can't let the perfect be the enemy of the good. It is more important to cover more concepts than to ensure that every concept exercise is flawless. I'll support adding `methods` to the concepts list for `health-statistics`.

Go ahead and claim the issue; I'm not aware of any competing work on it. Thanks for working on this!

**Comment by nahuakang** [comment]
@coriolinus Can you assign it to me? Can't really claim on issues :)

**Comment by nahuakang** [comment]
@coriolinus Just double-checking again that this is the link I need to create a new concept dir right? https://exercism.org/docs/building/tracks/concepts Thanks again :)

**Comment by coriolinus** [comment]
Yes, that's the correct documentation.
